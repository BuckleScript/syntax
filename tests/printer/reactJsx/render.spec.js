runPrinter(__dirname, false, false)
