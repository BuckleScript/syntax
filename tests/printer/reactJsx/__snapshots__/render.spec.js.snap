// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`commentAtTop.react.res 1`] = `
"@bs.obj
external makeProps: (
  ~msg: 'msg,
  ~key: string=?,
  unit,
) => {
  \\"msg\\": 'msg,
  // test React JSX file
} = \\"\\"

let make =
  (@warning(\\"-16\\") ~msg) => {
    ReactDOMRe.createDOMElementVariadic(\\"div\\", [{msg->React.string}])
  }
let make = {
  let \\\\\\"CommentAtTop.react\\" = (\\\\\\"Props\\": {\\"msg\\": 'msg}) =>
    make(~msg=\\\\\\"Props\\"[\\"msg\\"])
  \\\\\\"CommentAtTop.react\\"
}
"
`;

exports[`externalWithCustomName.react.res.fixme 1`] = `""`;

exports[`innerModule.react.res 1`] = `
"module Bar = {
  @bs.obj
  external makeProps: (
    ~a: 'a,
    ~b: 'b,
    ~key: string=?,
    unit,
  ) => {\\"a\\": 'a, \\"b\\": 'b} = \\"\\"
  let make =
    (@warning(\\"-16\\") ~a, @warning(\\"-16\\") ~b, _) => {
      Js.log(\\"This function should be named \`Test$Bar\`\\")
      ReactDOMRe.createDOMElementVariadic(\\"div\\", [])
    }
  let make = {
    let \\\\\\"InnerModule.react$Bar\\" = (\\\\\\"Props\\": {\\"a\\": 'a, \\"b\\": 'b}) =>
      make(~b=\\\\\\"Props\\"[\\"b\\"], ~a=\\\\\\"Props\\"[\\"a\\"], ())
    \\\\\\"InnerModule.react$Bar\\"
  }
  @bs.obj
  external componentProps: (
    ~a: 'a,
    ~b: 'b,
    ~key: string=?,
    unit,
  ) => {\\"a\\": 'a, \\"b\\": 'b} = \\"\\"

  let component =
    (@warning(\\"-16\\") ~a, @warning(\\"-16\\") ~b, _) => {
      Js.log(\\"This function should be named \`Test$Bar$component\`\\")
      ReactDOMRe.createDOMElementVariadic(\\"div\\", [])
    }
  let component = {
    let \\\\\\"InnerModule.react$Bar$component\\" = (\\\\\\"Props\\": {\\"a\\": 'a, \\"b\\": 'b}) =>
      component(~b=\\\\\\"Props\\"[\\"b\\"], ~a=\\\\\\"Props\\"[\\"a\\"], ())
    \\\\\\"InnerModule.react$Bar$component\\"
  }
}
"
`;

exports[`topLevel.react.res 1`] = `
"@bs.obj
external makeProps: (
  ~a: 'a,
  ~b: 'b,
  ~key: string=?,
  unit,
) => {\\"a\\": 'a, \\"b\\": 'b} = \\"\\"
let make =
  (@warning(\\"-16\\") ~a, @warning(\\"-16\\") ~b, _) => {
    Js.log(\\"This function should be named 'TopLevel.react'\\")
    ReactDOMRe.createDOMElementVariadic(\\"div\\", [])
  }
let make = {
  let \\\\\\"TopLevel.react\\" = (\\\\\\"Props\\": {\\"a\\": 'a, \\"b\\": 'b}) =>
    make(~b=\\\\\\"Props\\"[\\"b\\"], ~a=\\\\\\"Props\\"[\\"a\\"], ())
  \\\\\\"TopLevel.react\\"
}
"
`;
